runhouse/servers/cluster_servlet.py

```diff
@@ -382,42 +411,78 @@ async def _status_for_env_servlet(self, env_servlet_name):
         except Exception as e:
             return {"env_servlet_name": env_servlet_name, "Exception": e}
 
+    async def _aperiodic_gpu_check(self):
+        """periodically collects cluster gpu usage"""
+
+        pynvml.nvmlInit()  # init nvidia ml info collection
+
+        while True:
+            try:
+
+                gpu_count = pynvml.nvmlDeviceGetCount()
+                with self.lock:
+                    if not self.gpu_info:
+                        self.gpu_info = {device: [] for device in range(gpu_count)}
```

Comment:
what type is `device`? can we add type hints to the where this is intialized (ex: `self.gpu_info: Dict[int, Dict]`)